ang/civicase-base/providers/case-type-buttons.provider.js

```diff
@@ -0,0 +1,40 @@
+(function (angular, $, _) {
+  var module = angular.module('civicase-base');
+
+  module.provider('CaseTypeButtons', function () {
+    var caseTypeButtons = {};
+
+    this.addButtons = addButtons;
+    this.$get = $get;
+
+    /**
+     * Provides the case types.
+     *
+     * @returns {object[]} the list of case types.
+     */
+    function $get () {
+      return caseTypeButtons;
+    }
+
+    /**
+     * Adds the given buttons to their corresponding case types. These buttons will be shown
+     * in the list of case types on the dashboard.
+     *
+     * @typedef {{
+     *  url: string
+     * }} ButtonConfig
+     * @param {string} caseTypeName the name for the case type the buttons belong to.
+     * @param {ButtonConfig[]} buttonsConfig a list of case type buttons configurations.
```

Comment:
Is this really working? is this a correct doc format?
(I have no idea, if it works, could you share some links with me, where I can read more about these custom data types)

Reply:
It does work :)
![gif](https://user-images.githubusercontent.com/1642119/68021267-54787900-fc77-11e9-8707-6c3827ed7022.gif)

https://jsdoc.app/tags-typedef.html

Not sure how to share a type between files, so maybe I'd not go too crazy with it.